### PR TITLE
feat: linux support

### PR DESCRIPTION
**Pull request by nekowinston**
Hi there 👋 

I really enjoyed using your plugin on macOS, and recently, I had to spend a few weeks using Linux as my daily driver, so I gave adding Linux support a shot.

I've optimized a few things here and there, added types & doc. The Linux query command should work anywhere that has `xdg-portal`'s `org.freedesktop.appearance color-scheme` implemented. (GNOME, KDE, agnostic via [darkman](https://gitlab.com/WhyNotHugo/darkman))

Notable changes:
- `nvim_set_option` -> `nvim_set_option_value` (supported since at least 0.7, the former is [discouraged with 0.10](https://github.com/neovim/neovim/blob/642586977158b44c007177169d3091840a72a8fd/runtime/doc/news.txt#L204))
- `check_is_root`/`local root` -> `vim.loop.getuid()` supported since 0.8. nvim is already aware of the context it was started in, so there's no need to call an external command.
- same for `io.popen("uname -s"):read("*a")):match("Darwin")` -> `vim.loop.os_uname().sysname`
- `init()` is now called at the end of `setup()`, making both `opts = {...}` and `config = function() ...` for Lazy unnecessary, opts alone will do the job.
- `types.lua` for Neodev/LuaLS type hints
- auto-generated docs via panvimdoc

Happy to undo and/or refine some parts, just let me know.

### CODE REVIEW

**Review by f-person — APPROVED** [review]
This is amazing! Thank you! <3

File: lua/auto-dark-mode/init.lua [hunk 1]
```diff
@@ -80,14 +119,14 @@ local function disable()
 	vim.fn.timer_stop(timer_id)
 end
 
----@param options table<string, fun()>
----`options` contains two function - `set_dark_mode` and `set_light_mode`
+--`options` contains two function - `set_dark_mode` and `set_light_mode`
```
[the remark this answers is not in the document]
Comment (nekowinston) [reply lua/auto-dark-mode/init.lua]: Good catch, fixed

File: README.md [hunk 2]
```diff
@@ -1,23 +1,29 @@
 # auto-dark-mode.nvim
 A Neovim plugin for macOS that automatically changes the editor appearance
 based on system settings.
-![demo](assets/demo.gif?raw=true)
+
+<!-- panvimdoc-ignore-start -->
+<p align="center">
+    <img src="assets/demo.gif?raw=true" alt="demo"/>
```
[the remark this answers is not in the document]
Comment (nekowinston) [reply README.md:7]: I recorded a quick toggle demo, did you have a process when converting the other GIF? Maybe we should also enable Git LFS for ./assets/* files. 

https://github.com/f-person/auto-dark-mode.nvim/assets/79978224/2d90e0d4-9a01-4e46-93bb-9100023b89e2

[the remark this answers is not in the document]
Comment (f-person) [reply README.md:7]: > did you have a process when converting the other GIF?

nothing particular, probably used gifski

> Maybe we should also enable Git LFS for ./assets/* files.

that's a good idea! i never bothered with it but maybe it's time :)

[the remark this answers is not in the document]
Comment (f-person) [reply README.md:7]: the video looks great btw!

[the remark this answers is not in the document]
Comment (f-person) [reply README.md:7]: gifski convert of your video with these settings:
image


257661674-2d90e0d4-9a01-4e46-93bb-9100023b89e2

**Comment by f-person** [comment]
i'm working on the lfs migration (i think i'm done?). i'll merge this first and add the lfs changes, along with the linux gif in a new pr

**Comment by f-person** [comment]
@nekowinston the signing got messed up, perhaps because of rebasing (the github web ui should've handled it though?)
can you please open a new pr and i'll try merging instead?

**Comment by nekowinston** [comment]
Thanks for converting the video & merging it, it was already getting late for me. 😅 

I didn't know that LFS assets would cause issues for people (#10), sorry about that. 😨 
I figured plugin managers would ignore the files unless specified...

**Pull request merged**